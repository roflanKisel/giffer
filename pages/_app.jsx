import React from 'react';
import App, { Container } from 'next/app';
import { Provider } from 'react-redux';
import withRedux from 'next-redux-wrapper';
import withReduxSaga from 'next-redux-saga';
import { compose } from 'recompose';
import { injectGlobal } from 'styled-components';
import initializeStore from '../store/store';

class GifferApp extends App {
  static async getInitialProps({ Component, ctx }) {
    let pageProps = {};

    if (Component.getInitialProps) {
      pageProps = await Component.getInitialProps({ ctx });
    }

    return { pageProps };
  }

  render() {
    const { Component, pageProps, store } = this.props;

    return (
      <Container>
        <Provider store={store}>
          <Component {...pageProps} />
        </Provider>
      </Container>
    );
  }
}

injectGlobal`
  body {
    margin: 0px;
    font-family: Roboto;
  }
`;

export default compose(
  withRedux(initializeStore),
  withReduxSaga({ async: true })
)(GifferApp);
