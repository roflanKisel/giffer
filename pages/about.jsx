import React from 'react';
import styled from 'styled-components';
import withLayout from '../components/layout';

export const About = () => (
  <StyledParagraph>This page was created by Alexander Kiseliov</StyledParagraph>
);

const StyledParagraph = styled.p`
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
`;

export default withLayout(About);
