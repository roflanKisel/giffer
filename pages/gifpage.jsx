import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import withLayout from '../components/layout';
import GifArea from '../components/gif-area';

import { searchByIdRequest } from '../ducks/searchGifs';

export class Gifpage extends React.Component {
  static getInitialProps({ ctx: { query } }) {
    return { query };
  }

  render() {
    const {
      query,
      dispatchSearchByIdRequest,
      gif,
      isLoading,
      isFailure,
    } = this.props;

    return (
      <GifArea
        gif={gif}
        gifId={query.id}
        handleSearchById={dispatchSearchByIdRequest}
        isLoading={isLoading}
        isFailure={isFailure}
      />
    );
  }
}

Gifpage.defaultProps = {
  gif: null,
};

Gifpage.propTypes = {
  dispatchSearchByIdRequest: PropTypes.func.isRequired,
  query: PropTypes.shape({
    id: PropTypes.string,
  }).isRequired,
  gif: PropTypes.shape({
    url: PropTypes.string,
    height: PropTypes.string,
    width: PropTypes.string,
  }),
  isLoading: PropTypes.bool.isRequired,
  isFailure: PropTypes.bool.isRequired,
};

const mapStateToProps = state => ({
  isLoading: state.gifsData.isLoading,
  isFailure: state.gifsData.isFailure,
  gif: state.gifsData.searchedGif,
});

const mapDispatchToProps = dispatch => ({
  dispatchSearchByIdRequest: bindActionCreators(searchByIdRequest, dispatch),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withLayout(Gifpage));
