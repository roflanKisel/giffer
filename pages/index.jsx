import React from 'react';
import styled from 'styled-components';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import {
  searchByQueryRequest,
  trendingSearchRequest,
} from '../ducks/searchGifs';
import withLayout from '../components/layout';
import SearchLine from '../components/search-line';
import ContentArea from '../components/content-area';

const Index = ({
  isLoading,
  isFailure,
  contentList,
  dispatchSearchByQueryRequest,
  dispatchTrendingSearchRequest,
}) => (
  <StyledIndex>
    <SearchLine handleSearchByQuery={dispatchSearchByQueryRequest} />
    <ContentArea
      isLoading={isLoading}
      isFailure={isFailure}
      contentList={contentList}
      handleTrendingSearch={dispatchTrendingSearchRequest}
    />
  </StyledIndex>
);

const StyledIndex = styled.div`
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
  align-items: center;
`;

const mapStateToProps = state => ({
  isLoading: state.gifsData.isLoading,
  isFailure: state.gifsData.isFailure,
  contentList: state.gifsData.content,
});

const mapDispatchToProps = dispatch => ({
  dispatchSearchByQueryRequest: bindActionCreators(
    searchByQueryRequest,
    dispatch
  ),
  dispatchTrendingSearchRequest: bindActionCreators(
    trendingSearchRequest,
    dispatch
  ),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withLayout(Index));
