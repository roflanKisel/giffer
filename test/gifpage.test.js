import React from 'react';
import { shallow, mount } from 'enzyme';
import { Gifpage } from '../pages/gifpage';

describe('Gifpage', () => {
  const gifMock = {
    images: { original: { url: 'url', width: '200', height: '250' } },
  };

  const queryMock = { id: '1' };

  describe('Gifpage DUMB component', () => {
    it('should renders correctly when loading', () => {
      const wrapper = shallow(
        <Gifpage
          query={{ id: '1' }}
          dispatchSearchByIdRequest={() => {}}
          gif={null}
          isLoading={true}
          isFailure={false}
        />
      );

      expect(wrapper).toMatchSnapshot();
    });

    it('should renders correctly when failed', () => {
      const wrapper = shallow(
        <Gifpage
          query={queryMock}
          gif={null}
          dispatchSearchByIdRequest={() => {}}
          isLoading={false}
          isFailure={true}
        />
      );

      expect(wrapper).toMatchSnapshot();
    });

    it('should renders correctly when not failed', () => {
      const wrapper = shallow(
        <Gifpage
          dispatchSearchByIdRequest={() => {}}
          query={queryMock}
          gif={gifMock}
          isLoading={false}
          isFailure={false}
        />
      );

      expect(wrapper).toMatchSnapshot();
    });

    it('should receive all props', () => {
      const wrapper = mount(
        <Gifpage
          dispatchSearchByIdRequest={() => {}}
          query={queryMock}
          gif={gifMock}
          isLoading={false}
          isFailure={false}
        />
      );

      expect(wrapper.prop('dispatchSearchByIdRequest')).toBeTruthy();
      expect(wrapper.prop('query')).toEqual(queryMock);
      expect(wrapper.prop('gif')).toEqual(gifMock);
      expect(wrapper.prop('isLoading')).toEqual(false);
      expect(wrapper.prop('isFailure')).toEqual(false);
    });
  });
});
