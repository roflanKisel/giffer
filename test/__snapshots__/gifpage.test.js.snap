// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gifpage Gifpage DUMB component should renders correctly when failed 1`] = `
<lifecycle(Component)
  gif={null}
  gifId="1"
  handleSearchById={[Function]}
  isFailure={true}
  isLoading={false}
/>
`;

exports[`Gifpage Gifpage DUMB component should renders correctly when loading 1`] = `
<lifecycle(Component)
  gif={null}
  gifId="1"
  handleSearchById={[Function]}
  isFailure={false}
  isLoading={true}
/>
`;

exports[`Gifpage Gifpage DUMB component should renders correctly when not failed 1`] = `
<lifecycle(Component)
  gif={
    Object {
      "images": Object {
        "original": Object {
          "height": "250",
          "url": "url",
          "width": "200",
        },
      },
    }
  }
  gifId="1"
  handleSearchById={[Function]}
  isFailure={false}
  isLoading={false}
/>
`;
